src/Service.Tests/Configuration/ConfigurationTests.cs

```diff
@@ -1298,6 +1298,60 @@ public async Task TestConfigIsValid()
             }
         }
 
+        /// <summary>
+        /// Test to verify that provided value of depth-limit in the config file should be greater than 0.
+        /// -1 and null are special values.
+        /// -1 can be set to remove the depth limit, while `null` is the default value which means no depth limit check.
+        /// This test validates that depth-limit outside the valid range should fail validation
+        /// during `dab validate` and `dab start`.
+        /// </summary>
+        [DataTestMethod]
+        [DataRow(-1, true, DisplayName = "[PASS]: Valid Value: -1 to disable depth limit")]
```

Comment:
Bool function parameters are a code smell we should try to avoid. It makes the test harder to read/understand and harder to add in cases. it would be better to split this into 2 test cases, with shared code between them in a common helper function. One case for valid input, one case for invalid input.